huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -637,6 +693,69 @@ func createKafkaInstanceWithProductID(ctx context.Context, d *schema.ResourceDat
 	return nil
 }
 
+func waitForOrderComplete(ctx context.Context, d *schema.ResourceData, conf *config.Config,
+	client *golangsdk.ServiceClient, instanceID string) error {
+	region := conf.GetRegion(d)
+	orderId, err := waitForInstanceCreated(ctx, d, client, instanceID)
+	if err != nil {
+		return err
+	}
+	if orderId == "" {
+		log.Printf("[WARN] error get order id by instance ID: %s", instanceID)
+		return nil
+	}
+
+	bssClient, err := conf.BssV2Client(region)
+	if err != nil {
+		return fmt.Errorf("error creating BSS v2 client: %s", err)
+	}
+	// wait for order success
+	err = common.WaitOrderComplete(ctx, bssClient, orderId, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return err
+	}
+	_, err = common.WaitOrderResourceComplete(ctx, bssClient, orderId, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return fmt.Errorf("error waiting for Kafka order resource %s complete: %s", orderId, err)
+	}
+	return nil
+}
+
+func waitForInstanceCreated(ctx context.Context, d *schema.ResourceData, client *golangsdk.ServiceClient, instanceID string) (string, error) {
```

Comment:
how about rename this function to `getInstanceOrderId` ?

Reply:
it's a good advice, done